### PR TITLE
CLOUDP-120447: Telemetry - Track Events for Interactive Usages

### PR DESCRIPTION
**Pull request by fmenezes**
## Proposed changes

Telemetry - Track Events for Interactive Usages

_Jira ticket:_ CLOUDP-120447

Closes #[issue number]

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments

This PR is changing every call to `survey.Ask` and `survey.AskOne` with `telemetry.TrackAsk` and `telemetry.TrackAskOne`, these are drop-in replacements. Moving forward we should not use the `survey` functions above mentioned.

### CODE REVIEW

**Review by johansteyn — COMMENTED** [review]
Filipe, trying out your changes I just got a panic during config init after entering the public API key.

To reproduce:

1. Remove your config file
2. Run "atlas config init"
3. Enter a public key
4. Enter a private key

It should panic afetr step 4.

File: mongocli.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ You can place this binary somewhere in your `PATH` for ease of use.
 
 ```bash
 git clone https://github.com/mongodb/mongodb-atlas-cli.git
-cd mongocli
+cd mongodb-atlas-cli
```
Comment (fmenezes) [inline mongocli.md:37]: flyby

**Comment by fmenezes** [comment]
@johansteyn this should be ready for another look

File: internal/telemetry/event.go [hunk 2]
```diff
@@ -54,6 +54,25 @@ func withProfile() eventOpt { // either "default" or base64 hash
 	}
 }
 
+func withPrompt(p, k string) eventOpt {
+	return func(event Event) {
+		event.Properties["prompt"] = p
+		event.Properties["prompt_kind"] = k
```
Comment (johansteyn) [inline internal/telemetry/event.go:60]: Change "prompt_kind" to "prompt_type"

File: internal/telemetry/event_test.go [hunk 3]
```diff
@@ -301,11 +301,46 @@ func TestTelemetry_withError(t *testing.T) {
...
+	config.ToolName = config.AtlasCLI
+
+	q := "random question"
+	k := "select"
+
+	e := newEvent(withPrompt(q, k))
+
+	a := assert.New(t)
+	a.Equal(q, e.Properties["prompt"])
+	a.Equal(k, e.Properties["prompt_kind"])
```
Comment (johansteyn) [inline internal/telemetry/event_test.go:323]: Change "prompt_kind" to "prompt_type"

**Review by johansteyn — COMMENTED** [review]
When running "atlas config init" and entering a value for "Public API Key", the "empty" property has a value of true, which should be false.

**Review by johansteyn — APPROVED** [review]
Verified that the requested changes have been done and I'm getting the expected behaviour now, thanks!

**Pull request merged**